rgw/v2/tests/s3cmd/test_ratelimit_global.py

```diff
@@ -0,0 +1,242 @@
+    io_info_initialize.initialize(basic_io_structure.initial())
+    user_info = resource_op.create_users(no_of_users_to_create=config.user_count)[0]
+    user_name = user_info["user_id"]
+    rgw_service = RGWService()
+
+    ip_and_port = s3cmd_reusable.get_rgw_ip_and_port(ssh_con)
+    s3_auth.do_auth(user_info, ip_and_port)
+    # add rate limit capability to rgw user
+    caps_add = utils.exec_shell_cmd(
+        f"radosgw-admin caps add --uid {user_name} "
+        + "--caps='users=*;buckets=*;ratelimit=*'"
+    )
+    data = json.loads(caps_add)
+    caps = data["caps"]
+    log.info(f" User Caps are :{caps}")
+
+    max_read_bytes = config.bucket_max_read_bytes
+    max_read_ops = config.bucket_max_read_ops
+    max_write_bytes = config.bucket_max_write_bytes
+    max_write_ops = config.bucket_max_write_ops
+
+    max_read_bytes_kb = math.ceil(float(max_read_bytes) / 1024)
+    max_write_bytes_kb = math.ceil(float(max_write_bytes) / 1024)
+
+    # create bucket and set limits
+    bucket_name = utils.gen_bucket_name_from_userid(user_name, rand_no=0)
+
+    ssl = config.ssl
+    s3cmd_reusable.create_bucket(bucket_name, ssl)
+    log.info(f"Bucket {bucket_name} created")
+    limset = utils.exec_shell_cmd(
+        "radosgw-admin global ratelimit set --ratelimit-scope=bucket "
+        + f"--max-read-ops={max_read_ops} "
+        + f"--max-read-bytes={max_read_bytes} --max-write-bytes={max_write_bytes} "
+        + f"--max-write-ops={max_write_ops}"
+    )
+    log.info(f"Rate limits set globally on bucket")
+    limenable = utils.exec_shell_cmd(
+        f"radosgw-admin global ratelimit enable --ratelimit-scope=bucket"
+    )
+    log.info("Restarting for global rate limit to take effect")
+    restart_service = rgw_service.restart(ssh_con)
+    sleep(5)
+    if restart_service is False:
+        raise TestExecError("RGW service restart failed")
+    limget = utils.exec_shell_cmd(
+        f"radosgw-admin global ratelimit get --ratelimit-scope=bucket"
+    )
+    log.info(f"Rate limits enabled globally on bucket : {limget} ")
+
+    # test the read and write ops limit
+    log.info("Testing global read and write ops limit")
+    s3cmd_reusable.rate_limit_read(bucket_name, max_read_ops, ssl)
+
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
+
+    s3cmd_reusable.rate_limit_write(bucket_name, max_write_ops, ssl)
+
+    # sleep a minute to reset the ops limit
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
+
+    # test the read and write data limit
+    log.info("Testing global read and write data limits")
+    s3cmd_reusable.rate_limit_read(bucket_name, max_read_bytes_kb, ssl)
+
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
+    s3cmd_reusable.rate_limit_write(bucket_name, max_write_bytes_kb, ssl)
+
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
+
+    # Set the rate limits for the user and enable them
```

Comment:
Can setting rate limit at user level and enable  be written as a method in reusable

Reply:
@mkasturi18 Added a method to set and enable limit